docs/OpenShift GitOps CLI User Guide.md

```diff
@@ -0,0 +1,373 @@
+In the `core` mode (`--core` argument specified), the CLI talks directly to the Kubernetes API server set as per the `KUBECONFIG` environment variable or the default file `$HOME/.kube/config`. There is no need for users to login into the ArgoCD server for executing commands.
+
+### Basic syntax
+- Normal mode
+  ```
+  argocd [command or options] [arguments…​]
+  ```
+
+- Core mode
+  ```
+  KUBECONFIG=~/.kube/config argocd --core [command or options] [arguments…​]
+  ```
+### Global options
+
+| Option| Argument Type | Description|
+| ----- | ------------- |----------- |
+| --auth-token | string | Authentication token    |
+| --client-crt | string | Client certificate file
+| --client-crt-key | string | Client certificate key file |
+| --config | string | Path to Argo CD config (default "/home/user/.config/argocd/config") |
+| --controller-name | string | Name of the Argo CD Application controller; set this or the ARGOCD_APPLICATION_CONTROLLER_NAME environment variable when the controller's name label differs from the default, for example when installing via the Helm chart (default "argocd-application-controller") |
+| --core | | If set to true then CLI talks directly to Kubernetes instead of talking to Argo CD API server |
+| --grpc-web | | Enables gRPC-web protocol. Useful if Argo CD server is behind proxy which does not support HTTP2. |
+| --grpc-web-root-path | string | Enables gRPC-web protocol. Useful if Argo CD server is behind proxy which does not support HTTP2. Set web root.|
+| -H, --header | strings | Sets additional header to all requests made by Argo CD CLI. (Can be repeated multiple times to add multiple headers, also supports comma separated headers) |
+| -h, --help | | help for argocd |
+| --http-retry-max | int | Maximum number of retries to establish http connection to Argo CD server |
+| --insecure | | Skip server certificate and domain verification |
+| --kube-context | string | Directs the command to the given kube-context |
+| --logformat | string | Set the logging format. One of: text|json (default "text") |
+| --loglevel | string | Set the logging level. One of: debug|info|warn|error (default "info") |
+| --plaintext | | Disable TLS |
+| --port-forward | | Connect to a random argocd-server port using port forwarding |
+| --port-forward-namespace | string | Namespace name which should be used for port forwarding |
+| --redis-haproxy-name | string | Name of the Redis HA Proxy; set this or the ARGOCD_REDIS_HAPROXY_NAME environment variable when the HA Proxy's name label differs from the default, for example when installing via the Helm chart (default "argocd-redis-ha-haproxy") |
+| --redis-name | string | Name of the Redis deployment; set this or the ARGOCD_REDIS_NAME environment variable when the Redis's name label differs from the default, for example when installing via the Helm chart (default "argocd-redis") |
+
+
+### Global Commands
```

Comment:
@anandf Suggest using `Utility commmands` instead. Refer to https://docs.openshift.com/container-platform/4.10/cli_reference/tkn_cli/op-tkn-reference.html#op-tkn-utility-commands_op-tkn-reference for example.

Reply:
Changed title to Utility Commands as suggested.